### PR TITLE
expenses: init at 0.2.1

### PR DESCRIPTION
**Pull request by manojkarthick**
###### Motivation for this change

An interactive command line expense logger
GitHub: https://github.com/manojkarthick/expenses

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:5487]: Please put this into a separate commit with the message ``maintainers: add manojkarthick``.

File: pkgs/applications/misc/expenses/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/misc/expenses/default.nix:14]:
```suggestion
    owner = "manojkarthick";
    repo = "expenses";
    rev = "v${version}";
```

File: pkgs/applications/misc/expenses/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/misc/expenses/default.nix:20]: Please add a comment why you added this.

File: pkgs/applications/misc/expenses/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/misc/expenses/default.nix:25]:
```suggestion
    "-ldflags=-s -w -X github.com/manojkarthick/expenses/cmd.Version=${version}"
```

File: pkgs/applications/misc/expenses/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/misc/expenses/default.nix:28]:
```suggestion

```

File: pkgs/applications/misc/expenses/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/misc/expenses/default.nix:33]:
```suggestion
```

set by buildGoModule.

File: pkgs/applications/misc/expenses/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/misc/expenses/default.nix:31]:
```suggestion
   license = licenses.mit;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by manojkarthick** [comment]
Thanks for the review @SuperSandro2000 - I've updated the PR with the requested changes.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108753` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>expenses</li>
</ul>
</details>

File: pkgs/top-level/all-packages.nix
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:21433]:
```suggestion
  expenses = callPackage ../applications/misc/expenses { };
```

Comment (manojkarthick) [reply pkgs/top-level/all-packages.nix]: @SuperSandro2000 Updated

**Comment by manojkarthick** [comment]
@SuperSandro2000 - All the comments have been resolved, and it seems to be building okay. Would you be able to approve and merge this when you get a chance? Thanks!

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108753` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>expenses</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108753` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>expenses</li>
</ul>
</details>

**Comment by manojkarthick** [comment]
Thank you!

**Pull request merged**
